src/main/java/hudson/tasks/test/TestResult.java

```diff
@@ -132,7 +136,8 @@ public Result getBuildResult() {
      */
     @Override
     public TestResult getPreviousResult() {
-        Run<?, ?> b = getRun();
+        Run<?, ?> originBuild = getRun();
+        Run<?, ?> b = originBuild;
```

Comment:
why do you need both variables?

Reply:
Needed it to look up job (as I can't really trust past builds here), but I just reworked it in 254298b

Reply:
much better, thanks